modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -254,10 +264,16 @@ public ObjectField addSystemObjectField(
 			dbColumnName = name;
 		}
 
+		String readOnly = ObjectFieldConstants.READ_ONLY_FALSE;
```

Comment:
This code is bypassing the feature flag, because of the system ObjectFields behavior, I suggest this way:

```java
		String readOnly = null;

		if (FeatureFlagManagerUtil.isEnabled("LPS-170122")) {
			readOnly = ObjectFieldConstants.READ_ONLY_FALSE;

			if (_readOnlyObjectFieldNames.contains(name)) {
				readOnly = ObjectFieldConstants.READ_ONLY_TRUE;
			}
		}
```